To: vim_dev@googlegroups.com
Subject: Patch 7.3.471
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.471
Problem:    Can't abort listing placed signs.
Solution:   Check "got_int". (Christian Brabandt)
Files:	    src/buffer.c, src/ex_cmds.c


*** ../vim-7.3.470/src/buffer.c	2012-03-07 22:52:30.000000000 +0100
--- src/buffer.c	2012-03-16 13:54:48.000000000 +0100
***************
*** 5557,5563 ****
  	buf = firstbuf;
      else
  	buf = rbuf;
!     while (buf != NULL)
      {
  	if (buf->b_signlist != NULL)
  	{
--- 5557,5563 ----
  	buf = firstbuf;
      else
  	buf = rbuf;
!     while (buf != NULL && !got_int)
      {
  	if (buf->b_signlist != NULL)
  	{
***************
*** 5565,5571 ****
  	    MSG_PUTS_ATTR(lbuf, hl_attr(HLF_D));
  	    msg_putchar('\n');
  	}
! 	for (p = buf->b_signlist; p != NULL; p = p->next)
  	{
  	    vim_snprintf(lbuf, BUFSIZ, _("    line=%ld  id=%d  name=%s"),
  			   (long)p->lnum, p->id, sign_typenr2name(p->typenr));
--- 5565,5571 ----
  	    MSG_PUTS_ATTR(lbuf, hl_attr(HLF_D));
  	    msg_putchar('\n');
  	}
! 	for (p = buf->b_signlist; p != NULL && !got_int; p = p->next)
  	{
  	    vim_snprintf(lbuf, BUFSIZ, _("    line=%ld  id=%d  name=%s"),
  			   (long)p->lnum, p->id, sign_typenr2name(p->typenr));
*** ../vim-7.3.470/src/ex_cmds.c	2012-03-07 14:57:50.000000000 +0100
--- src/ex_cmds.c	2012-03-16 13:58:50.000000000 +0100
***************
*** 6729,6735 ****
  	if (idx == SIGNCMD_LIST && *arg == NUL)
  	{
  	    /* ":sign list": list all defined signs */
! 	    for (sp = first_sign; sp != NULL; sp = sp->sn_next)
  		sign_list_defined(sp);
  	}
  	else if (*arg == NUL)
--- 6729,6735 ----
  	if (idx == SIGNCMD_LIST && *arg == NUL)
  	{
  	    /* ":sign list": list all defined signs */
! 	    for (sp = first_sign; sp != NULL && !got_int; sp = sp->sn_next)
  		sign_list_defined(sp);
  	}
  	else if (*arg == NUL)
*** ../vim-7.3.470/src/version.c	2012-03-07 22:55:17.000000000 +0100
--- src/version.c	2012-03-16 13:55:28.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     471,
  /**/

-- 
Some of the well know MS-Windows errors:
	EMEMORY		Memory error caused by..., eh...
	ELICENSE	Your license has expired, give us more money!
	EMOUSE		Mouse moved, reinstall Windows
	EILLEGAL	Illegal error, you are not allowed to see this
	EVIRUS		Undetectable virus found

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
